plugins/ingest-attachment/src/main/java/org/elasticsearch/ingest/attachment/AttachmentProcessor.java

```diff
@@ -161,59 +191,99 @@ String getTargetField() {
         return targetField;
     }
 
-    Set<Property> getProperties() {
+    Set<String> getProperties() {
         return properties;
     }
 
+    public Set<ReservedProperty> getReservedProperties() {
+        return reservedProperties;
+    }
+
     int getIndexedChars() {
         return indexedChars;
     }
 
     public static final class Factory implements Processor.Factory {
 
-        static final Set<Property> DEFAULT_PROPERTIES = EnumSet.allOf(Property.class);
+        static final Set<ReservedProperty> DEFAULT_PROPERTIES = EnumSet.allOf(ReservedProperty.class);
 
         @Override
         public AttachmentProcessor create(Map<String, Processor.Factory> registry, String processorTag,
                                           Map<String, Object> config) throws Exception {
             String field = readStringProperty(TYPE, processorTag, config, "field");
             String targetField = readStringProperty(TYPE, processorTag, config, "target_field", "attachment");
-            List<String> properyNames = readOptionalList(TYPE, processorTag, config, "properties");
+            List<String> propertyNames = readOptionalList(TYPE, processorTag, config, "properties");
             int indexedChars = readIntProperty(TYPE, processorTag, config, "indexed_chars", NUMBER_OF_CHARS_INDEXED);
             boolean ignoreMissing = readBooleanProperty(TYPE, processorTag, config, "ignore_missing", false);
 
-            final Set<Property> properties;
-            if (properyNames != null) {
-                properties = EnumSet.noneOf(Property.class);
-                for (String fieldName : properyNames) {
-                    try {
-                        properties.add(Property.parse(fieldName));
-                    } catch (Exception e) {
-                        throw newConfigurationException(TYPE, processorTag, "properties", "illegal field option [" +
-                            fieldName + "]. valid values are " + Arrays.toString(Property.values()));
+            final Set<ReservedProperty> reservedProperties;
+            final Set<String> properties = new HashSet<>();
+            if (propertyNames != null) {
+                reservedProperties = EnumSet.noneOf(ReservedProperty.class);
+                for (String fieldName : propertyNames) {
+                    // We build the regex automaton we will use to extract raw metadata
+                    CharacterRunAutomaton automaton = new CharacterRunAutomaton(simpleMatchToAutomaton(fieldName));
```

Comment:
the transport reindex operation also calls `MinimizationOperations.minimize(automaton, Operations.DEFAULT_MAX_DETERMINIZED_STATES);` after the automaton creation, which looks useful to me